### PR TITLE
712 - `{shinytest2}` for `tm_t_crosstable`

### PR DESCRIPTION
**Pull request by kartikeyakirar**
Part of https://github.com/insightsengineering/teal.modules.general/issues/712

### CODE REVIEW

File: tests/testthat/test-shinytest2-tm_t_crosstable.R [hunk 1]
```diff
@@ -0,0 +1,106 @@
+app_driver_tm_t_crosstable <- function() {
+  init_teal_app_driver(
+    data = simple_cdisc_data(),
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply tests/testthat/test-shinytest2-tm_t_crosstable.R:3]: thanks.

**Comment by m7pr** [comment]
@kartikeyakirar your `app_driver_tm_t_crosstable` function is missing `teal.transform::` prefixes (and maybe `teal.widgets::` if you use `ggplot2_args`

**Comment by kartikeyakirar** [comment]
> @kartikeyakirar your app_driver_tm_t_crosstable function is missing teal.transform:: prefixes (and maybe teal.widgets:: if you use ggplot2_args

done [94cb2f1](https://github.com/insightsengineering/teal.modules.general/pull/733/commits/94cb2f10f41182897aef8296d6342702292b5a92)

File: tests/testthat/test-shinytest2-tm_t_crosstable.R [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    app_driver$get_text("#teal-main_ui-root-active_tab > li.active > a"),
+    "Cross Table"
+  )
+
+  encoding_dataset <- app_driver$get_text("#teal-main_ui-root-cross_table .help-block")
+  testthat::expect_match(encoding_dataset, "Dataset:\\n *ADSL\\n", all = FALSE)
+
+  app_driver$stop()
+})
+
```
Comment (averissimo) [inline tests/testthat/test-shinytest2-tm_t_crosstable.R:62]: This test seems optional, given that we are no longer doing it on the other PRS (relying on example testing and initialization of other tests)

But I'm ok with keeping it if you feel it's necessary.

**Review by averissimo — APPROVED** [review]
LGTM!

File: tests/testthat/test-shinytest2-tm_t_crosstable.R (hunk 2 above)
Comment (kartikeyakirar) [reply tests/testthat/test-shinytest2-tm_t_crosstable.R:62]: Lets keep this  for now.

**Comment by m7pr** [comment]
One issue

```r
  Loading required package: teal.transform
  [ FAIL 1 | WARN 5 | SKIP 0 | PASS 366 ]
  
  ══ Failed tests ════════════════════════════════════════════════════════════════
  ── Failure ('test-shinytest2-tm_misssing_data.R:112:3'): e2e - tm_missing_data: Check default settings and visibility of the combinations graph and encodings ──
  app_driver$get_active_module_input("iris-combination_cutoff") not equal to 2L.
  1/1 mismatches
  [1] 1 - 2 == -1
```

**Comment by kartikeyakirar** [comment]
I am not sure what's happening there. The default value for `app_driver$get_active_module_input("iris-combination_cutoff")` is 2, and the checks passed here https://github.com/insightsengineering/teal.modules.general/pull/727. 🤔

**Pull request merged**